protocols/v2/subprotocols/mining/src/set_new_prev_hash.rs

```diff
@@ -6,27 +6,28 @@ use binary_sv2::{Deserialize, Serialize, U256};
 #[cfg(not(feature = "with_serde"))]
 use core::convert::TryInto;
 
-/// # SetNewPrevHash (Server -> Client, broadcast)
+/// Message used by upstream to share or distribute the latest block hash.
 ///
-/// Prevhash is distributed whenever a new block is detected in the network by an upstream node.
-/// This message MAY be shared by all downstream nodes (sent only once to each channel group).
-/// Clients MUST immediately start to mine on the provided prevhash. When a client receives this
-/// message, only the job referenced by Job ID is valid. The remaining jobs already queued by the
-/// client have to be made invalid.
-/// Note: There is no need for block height in this message.
+/// This message may be shared by all downstream nodes (sent only once to each channel group).
+///
+/// Downstream must immediately start to mine on the provided [`SetNewPrevHash::prevhash`].
+///
+/// When a downstream receives this message, only the job referenced by [`SetNewPrevHash::job_id`]
+/// is valid. Remaining jobs have to be dropped.
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct SetNewPrevHash<'decoder> {
     /// Group channel or channel that this prevhash is valid for.
     pub channel_id: u32,
-    /// ID of a job that is to be used for mining with this prevhash. A pool may
-    /// have provided multiple jobs for the next block height (e.g. an empty
-    /// block or a block with transactions that are complementary to the set of
-    /// transactions present in the current block template).
+    /// Job identfier that is to be used for mining.
+    ///
+    /// A pool may have provided multiple jobs for the next block height (e.g. an empty block or a
+    /// block with transactions that are complementary to the set of transactions present in the
+    /// current block template).
     pub job_id: u32,
-    /// Previous block’s hash, block header field.
+    /// Latest block hash observed by the pool.
```

Comment:
we shouldn't assume only pool matters

```suggestion
    /// Latest block hash observed by the Template Provider.
```

Reply:
I am not sure why changing pool to template provider?

Reply:
where does prevhash come from?

Reply:
Yeah, to cover also JD case, I think that the best way to frame it is:
```suggestion
    /// Latest block hash observed by the Template Provider.
```

Reply:
@jbesraa did you understand this?

if yes, we can apply the suggestion by @GitGab19 and move forward to merging